### PR TITLE
Feature/matching algorithm

### LINKED ISSUE #181: Create and test the matching algortihm

### PR DESCRIPTION
**Pull request by alexs-rufer**
## 🎯 PR Description

Hey everyone! This PR introduces a **`TutorRecommender`** utility (the "matching algorithm") that ranks tutors based on academic level compatibility, section match, and tutor rating. Additionally, I've updated the `SuitabilityScoreCalculator` to expose the `computeAcademicLevelCompatibility` function, since we need it directly in `TutorRecommender`. 

The `TutorMatchingScreen` now applies the same initial filtering as before (subject, price range, availability, etc.) and then leverages `TutorRecommender` to further rank the already filtered tutors, resulting in more relevant tutor recommendations.

Also, I've removed the unused `Comment.kt` file since it’s no longer needed. Finally, I added `TutorRecommenderTest.kt` to ensure our ranking logic is solid.

No screenshots this time, as it’s mostly behind-the-scenes logic and tests!

❗ **Note**: Honestly, I’m not entirely sure how useful this final ranking is. Most parameters are basically “eliminatory” conditions—if you don’t meet them, you’re not considered at all—and the remaining parameters like rating or section match might not be the most meaningful differentiators. That said, it’s a start, and we at least have a more systematic approach now.

## 🧭 How Tutors Are Sorted

After the initial filtering, the remaining tutors are scored and sorted by the `TutorRecommender`. The scoring formula combines three main factors:

1. **Academic Level Compatibility (35%)**: Measures how closely the tutor’s academic level aligns with or surpasses the student’s level.
2. **Section Match (25%)**: Rewards tutors from the same academic section to ensure more relevant domain expertise.
3. **Tutor Rating (40%)**: Incorporates the average rating of completed lessons. If there are no rated lessons yet, new or unrated tutors get a full rating score (1.0), giving them a better chance to appear at the top and be discovered.

## 📂 Files Changed

**`SuitabilityScoreCalculator.kt`**
- Made `computeAcademicLevelCompatibility` public to be reused by `TutorRecommender`.

**`tutorRecommender.kt`**
- **New File**: Implements logic to compute a suitability score for tutors and returns them sorted based on academic level, section compatibility, and rating history.

**`TutorMatching.kt`**
- Integrated `TutorRecommender` after the existing filters to refine tutor recommendations.

**`Comment.kt`**
- Deleted since it's no longer needed.

**`TutorRecommenderTest.kt`**
- **New Test File**: Ensures `TutorRecommender` works as intended with different scenarios.

## 🧪 How to Test

Run the app, select or create a lesson, and check the recommended tutors list—tutors should now be more accurately ranked. Also, run `TutorRecommenderTest` to confirm everything is working well!

---

Close #181

### CODE REVIEW

File: app/src/main/java/com/github/se/project/model/lesson/tutorRecommender.kt
Comment (AlexisAllmnd) [inline app/src/main/java/com/github/se/project/model/lesson/tutorRecommender.kt:1]: Very nice algorithm that help student f
choosing the best tutor. Great job

File: app/src/main/java/com/github/se/project/model/lesson/tutorRecommender.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    val W2 = 0.25 // Section Match
+    val W3 = 0.4 // Tutor Rating
+
+    val tutorsWithScores =
+        availableTutors.map { tutorProfile ->
+          val X1 = computeAcademicLevelCompatibility(tutorProfile, studentProfile)
+          val X2 = computeSectionMatch(tutorProfile, studentProfile)
+          val X3 =
+              if (ratedLessons.isNotEmpty()) computeRatingScore(averageRating)
+              else 1.0 // priority to new tutors
```
Comment (AlexisAllmnd) [inline app/src/main/java/com/github/se/project/model/lesson/tutorRecommender.kt:38]: I like this philosophy, we give a chance for new tutor to be on the top of the screen for their first lesson

File: app/src/main/java/com/github/se/project/model/profile/Comment.kt
Comment (AlexisAllmnd) [inline app/src/main/java/com/github/se/project/model/profile/Comment.kt:1]: nice cleaning

File: app/src/test/java/com/github/se/project/model/lesson/TutorRecommenderTest.kt
Comment (AlexisAllmnd) [inline app/src/test/java/com/github/se/project/model/lesson/TutorRecommenderTest.kt:1]: Nice test for you algorithm !

**Review by AlexisAllmnd — APPROVED** [review]
Good job! You've significantly improved the matching algorithm by proposing the most relevant tutors for the student in question with solid and well-tested code. You've created clear and modular code, which will make it easy to update this algorithm if necessary in the future.

File: app/src/main/java/com/github/se/project/model/lesson/tutorRecommender.kt (hunk 1 above)
Comment (alexs-rufer) [reply app/src/main/java/com/github/se/project/model/lesson/tutorRecommender.kt:38]: thanks!

**Pull request merged**